XiEditor/EditViewController.swift

```diff
@@ -165,16 +169,32 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
         scrollView.hasHorizontalScroller = true
         scrollView.usesPredominantAxisScrolling = true
         (scrollView.contentView as? XiClipView)?.delegate = self
+
     }
 
     override func viewDidAppear() {
         super.viewDidAppear()
+        setupStatusBar()
         shadowView.setup()
         NotificationCenter.default.addObserver(self, selector: #selector(EditViewController.frameDidChangeNotification(_:)), name: NSView.frameDidChangeNotification, object: scrollView)
         // call to set initial scroll position once we know view size
         redrawEverything()
     }
 
+    func setupStatusBar() {
+        statusBar = StatusBar(frame: .zero)
+        statusBar.hasUnifiedTitlebar = unifiedTitlebar
+        self.view.addSubview(statusBar)
+
+        NSLayoutConstraint.activate([
+            statusBar.heightAnchor.constraint(equalToConstant: statusBar.statusBarHeight),
+            statusBar.widthAnchor.constraint(equalTo: editView.widthAnchor),
```

Comment:
just noticed this, but the width constraint shouldn't be necessary if you have a leading and trailing.